src/loader.js

```diff
@@ -40,3 +45,345 @@ export function createLoaderElement(doc, elementName) {
         <div class="i-amphtml-loader-dot"></div>
       </div>`;
 }
+
+/* NEW LOADER */
+
+/**
+ * Creates a default "loading indicator" element based on the new design.
+ *
+ * Please see https://github.com/ampproject/amphtml/issues/20237 for details,
+ * screenshots and various states of the new loader design.
+ *
+ * @param {!AmpElement} element
+ * @param {number} elementWidth
+ * @param {number} elementHeight
+ * @return {!Element} New loader root element
+ */
+export function createNewLoaderElement(element, elementWidth, elementHeight) {
+  devAssert(
+    isNewLoaderExperimentEnabled(toWin(element.ownerDocument.defaultView))
+  );
+
+  const loader = new LoaderBuilder(element, elementWidth, elementHeight);
+  return loader.build();
+}
+
+/**
+ * Helper class to build the new loader's DOM.
+ */
+class LoaderBuilder {
+  /**
+   * @param {!AmpElement} element
+   * @param {number} elementWidth
+   * @param {number} elementHeight
+   */
+  constructor(element, elementWidth, elementHeight) {
+    /** @private @const {!AmpElement} */
+    this.element_ = element;
+
+    /** @private @const  {number} */
+    this.layoutWidth_ = elementWidth;
+
+    /** @private @const  {number} */
+    this.layoutHeight_ = elementHeight;
+
+    /** @private {?Element} */
+    this.domRoot_;
+
+    /** @private {?Element} */
+    this.svgRoot_;
+  }
+
+  /**
+   * Builds the loader's DOM and returns the element.
+   * @return {!Element} new loader root element
+   */
+  build() {
+    this.buildContainers_();
+    this.maybeAddDefaultPlaceholder_();
+    this.maybeAddLoaderAnimation_();
+
+    return dev().assertElement(this.domRoot_);
+  }
+
+  /**
+   * Builds the wrappers for the loader.
+   * @private
+   */
+  buildContainers_() {
+    const html = htmlFor(this.element_);
+    this.domRoot_ = html`
+      <div class="i-amphtml-new-loader">
+        <div>
+          <svg
+            ref="svgRoot"
+            xmlns="http://www.w3.org/2000/svg"
+            viewBox="24 24 72 72"
+          ></svg>
+        </div>
+      </div>
+    `;
+
+    /**
+     * There is an extra inner div here for backward compatibility with
+     * customizing loaders. The common and documented CSS for customizing
+     * loaders includes a style to hide the old three dots via:
+     *  .my-custom-loader .amp-active > div {
+     *     display: none;
+     *  }
+     * The extra div mimic a similar DOM.
+     */
+    this.svgRoot_ = dev().assertElement(htmlRefs(this.domRoot_)['svgRoot']);
```

Comment:
Nit: Now that there's only one ref, we could drop the ref entirely and just use `querySelector('svg')`.